### PR TITLE
Test that @@unscopables is looked up once for inc/dec.

### PR DESCRIPTION
**Pull request by syg**
See https://bugzilla.mozilla.org/show_bug.cgi?id=1341061

### CODE REVIEW

**Comment by littledan** [comment]
I hate the world.

You could make this test marginally stronger by checking that flag is false after both test cases (which would verify that `@@unscopables` isn't read a second time, even if your test already already verifies that only the first read is used). You could also have a test case where it actually is reflected in `@@unscopables`, to verify that implementations do pay attention to unscopables getters (not sure if we have a separate test for that).

Either way, those are potential follow-ons, but this looks good to land already. LGTM.

**Comment by syg** [comment]
@littledan What's meant by "it actually is reflected in` @@unscopables`"?

**Comment by littledan** [comment]
@syg sorry, I meant, "where the `@@unscopables` value for the key was `true`, causing it to be excluded"

**Comment by littledan** [comment]
New version LGTM, that other test possibility is sort of a tangent.

File: test/language/statements/with/unscopables-inc-dec.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2017 Mozilla Corporation. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description:
```
Comment (leobalter) [inline test/language/statements/with/unscopables-inc-dec.js]: can you add a `>` at the end of this line to make it compatible with monkeyYaml?

`description: >`

Ref https://github.com/tc39/test262/pull/862

File: test/language/statements/with/unscopables-inc-dec.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  GetIdentifierReference ( lex, name, strict )
+  [...]
+  3. Let exists be ? envRec.HasBinding(name).
+
+  HasBinding ( N )
+  [...]
+  6. Let unscopables be ? Get(bindings, @@unscopables).
+flags: [noStrict]
+features: [Symbol.unscopables]
+---*/
```
Comment (leobalter) [inline test/language/statements/with/unscopables-inc-dec.js]: this is missing an `esid` tag

File: test/language/statements/with/unscopables-inc-dec.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                    } }) {
+    x++;
+  }
+}
+
+assert.sameValue(unscopablesGetterCalled, 1);
+assert.sameValue(a.x, 7);
+assert.sameValue(b.x, 5);
+
+flag = true;
```
Comment (leobalter) [inline test/language/statements/with/unscopables-inc-dec.js]: I like to set tests atomic, this means you could just reset the value of `unscopablesGetterCalled` to `0` here as well.

**Comment by syg** [comment]
Comments addressed.

**Comment by leobalter** [comment]
Thanks!

**Pull request merged**